examples/tutorials/physics_benchmarking.py

```diff
@@ -0,0 +1,334 @@
+    test_log += (
+        "\t ========================= Performance ======================== " + "\n"
+    )
+    test_log += "\t| Number of frames: " + str(len(physics_step_times)) + "\n"
+    test_log += (
+        "\t| Average time per physics step: "
+        + str(sum(physics_step_times) / len(physics_step_times))
+        + "\n"
+    )
+    test_log += (
+        "\t| Average time per frame render: "
+        + str(sum(graphics_render_times) / len(graphics_render_times))
+        + "\n"
+    )
+    test_log += "\t| Maximum Collisions: " + str(max(collisions)) + "\n"
+    test_log += "\t| Total time for test: " + str(end_time - start_time) + "\n"
+    test_log += "\t|----- END of %s -----" % test_id + "\n"
+    test_log += (
+        "\t ============================================================== " + "\n"
+    )
+    test_log += "\n"
+
+    print(test_log)
+    if make_video:
+        vut.make_video(
+            observations,
+            "rgba_camera_1stperson",
+            "color",
+            output_path + test_id,
+            open_vid=show_video,
+        )
+
+    for graph_name in graphs_to_produce:
+        if graph_name in data:
+            create_frame_line_plot(test_id, data[graph_name], graph_name)
+
+    return test_log
+
+
+def main():
+    log_contents = ""
+    if len(tests_to_run) == 0:
+        # runs all benchmarks when none are specified
+        for test_id in benchmarks:
+            log_contents += run_test(test_id)
+    else:
+        # only runs the benchmark/benchmark_sets specified in commandline arguments
+        for test in tests_to_run:
+            if test in benchmark_sets:
+                for sub_test in benchmark_sets[test]:
+                    log_contents += run_test(sub_test)
+            elif test in benchmarks:
+                log_contents += run_test(test)
+            else:
+                print(test, "done not exist.")
+                log_contents += test + " does not exist." + "\n" + "\n"
+
+    if save_log:
+        with open(output_path + "benchmark_log_file.txt", "w") as log:
+            log.write(log_contents)
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--no-show-video", dest="show_video", action="store_false")
+    parser.add_argument("--no-make-video", dest="make_video", action="store_false")
+    parser.add_argument("--save-to-log", dest="save_log", action="store_true")
+    parser.add_argument("--graph", default=[], nargs="*", type=str, metavar="N")
+    parser.add_argument("--run", default=[], nargs="*", type=str, metavar="N")
+
+    parser.set_defaults(show_video=True, make_video=True, save_log=False)
+    args, _ = parser.parse_known_args()
+
+    show_video = args.show_video
+    make_video = args.make_video
+    save_log = args.save_log
+    graphs_to_produce = args.graph
+    tests_to_run = args.run
+
+    if make_video and not os.path.exists(output_path):
```

Comment:
Will also want to produce the output path for logging and graph saving. Maybe just always create it?
```suggestion
    if not os.path.exists(output_path):
```